packages/flutter/test/cupertino/nav_bar_test.dart

```diff
@@ -58,48 +58,6 @@ void main() {
     expect(tester.getCenter(find.text('Page 2')).dx, 400.0);
   });
 
-  testWidgets('Opaque background does not add blur effects, non-opaque background adds blur effects', (WidgetTester tester) async {
-    const CupertinoDynamicColor background = CupertinoDynamicColor.withBrightness(
-      color: Color(0xFFE5E5E5),
-      darkColor: Color(0xF3E5E5E5),
-    );
-
-    await tester.pumpWidget(
-      const CupertinoApp(
-        theme: CupertinoThemeData(brightness: Brightness.light),
-        home: CupertinoNavigationBar(
-          middle: Text('Title'),
-          backgroundColor: background,
-        ),
-      ),
-    );
-    expect(find.byType(BackdropFilter), findsNothing);
-    expect(find.byType(CupertinoNavigationBar), paints..rect(color: background.color));
-
-    await tester.pumpWidget(
-      const CupertinoApp(
-        theme: CupertinoThemeData(brightness: Brightness.dark),
-        home: CupertinoNavigationBar(
-          middle: Text('Title'),
-          backgroundColor: background,
-        ),
-      ),
-    );
-    expect(find.byType(BackdropFilter), findsOneWidget);
-    expect(find.byType(CupertinoNavigationBar), paints..rect(color: background.darkColor));
-  });
-
-  testWidgets('Non-opaque background adds blur effects', (WidgetTester tester) async {
```

Comment:
I think we want to make sure that non-opaque backgrounds, with the automatic transparency turned off still apply a blur. It looks like your tests cover the other cases. Since we always have a BackdropFilter, can we check `BackdropFilter.filter`? And see if it's not a Blur with no sigma. At worst with `toString`?